core/src/main/java/com/linecorp/armeria/client/ClientFactoryBuilder.java

```diff
@@ -381,40 +464,25 @@ public ClientFactory build() {
                     firstNonNull(domainNameResolverCustomizers, ImmutableList.of()));
         }
 
+        @SuppressWarnings("unchecked")
+        final AddressResolverGroup<InetSocketAddress> addressResolverGroup =
+                (AddressResolverGroup<InetSocketAddress>) addressResolverGroupFactory.apply(workerGroup);
+
         return new DefaultClientFactory(new HttpClientFactory(
-                workerGroup, shutdownWorkerGroupOnClose, channelOptions, sslContextCustomizer,
-                addressResolverGroupFactory, http2InitialConnectionWindowSize, http2InitialStreamWindowSize,
+                workerGroup, shutdownWorkerGroupOnClose, eventLoopScheduler, channelOptions,
+                sslContextCustomizer, addressResolverGroup,
+                http2InitialConnectionWindowSize, http2InitialStreamWindowSize,
                 http2MaxFrameSize, http2MaxHeaderListSize, http1MaxInitialLineLength, http1MaxHeaderSize,
                 http1MaxChunkSize, idleTimeoutMillis, useHttp2Preface,
                 useHttp1Pipelining, connectionPoolListener, meterRegistry));
     }
 
     @Override
     public String toString() {
-        return toString(this, workerGroup, shutdownWorkerGroupOnClose, channelOptions,
-                        sslContextCustomizer, addressResolverGroupFactory, http2InitialConnectionWindowSize,
-                        http2InitialStreamWindowSize, http2MaxFrameSize, http2MaxHeaderListSize,
-                        http1MaxInitialLineLength, http1MaxHeaderSize, http1MaxChunkSize, idleTimeoutMillis,
-                        useHttp2Preface, useHttp1Pipelining, connectionPoolListener, meterRegistry);
-    }
-
-    static String toString(
-            ClientFactoryBuilder self,
-            EventLoopGroup workerGroup, boolean shutdownWorkerGroupOnClose,
-            Map<ChannelOption<?>, Object> socketOptions,
-            Consumer<? super SslContextBuilder> sslContextCustomizer,
-            @Nullable
-            Function<? super EventLoopGroup,
-                     ? extends AddressResolverGroup<? extends InetSocketAddress>> addressResolverGroupFactory,
-            int http2InitialConnectionWindowSize, int http2InitialStreamWindowSize, int http2MaxFrameSize,
-            long http2MaxHeaderListSize, int http1MaxInitialLineLength, int http1MaxHeaderSize,
-            int http1MaxChunkSize, long idleTimeoutMillis, boolean useHttp2Preface, boolean useHttp1Pipelining,
-            ConnectionPoolListener connectionPoolListener,
-            MeterRegistry meterRegistry) {
-
-        final ToStringHelper helper = MoreObjects.toStringHelper(self).omitNullValues();
+        final ToStringHelper helper = MoreObjects.toStringHelper(this).omitNullValues();
         helper.add("workerGroup", workerGroup + " (shutdownOnClose=" + shutdownWorkerGroupOnClose + ')')
-              .add("socketOptions", socketOptions)
+              .add("eventLoopSchedulerFactory", eventLoopSchedulerFactory)
```

Comment:
Perhaps we should add this only when the factory is not the default?

Reply:
`eventLoopSchedulerFactory` is nullable and we check the mutual exclusion in the builder, so I thought this was OK.
But let me check the `null` for clarity. 😉